include/zephyr/net/ieee802154_radio_ext_ot.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *   This file extends interface of ieee802154_radio.h for OpenThread.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_EXT_OT_H_
+#define ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_EXT_OT_H_
+
+#include <zephyr/net/ieee802154_radio.h>
+
+/**
+ *  OpenThread specific capabilities of ieee802154 driver.
+ *  This type extends ieee802154_hw_caps.
+ */
+enum ieee802154_ext_ot_hw_caps {
+	IEEE802154_EXT_OT_HW_MULTIPLE_CCA = BIT(IEEE802154_HW_CAPS_BITS_PRIV_START),
+};
+
+enum ieee802154_ext_ot_tx_mode {
+	/**
+	 * Transmit packet in the future, perform one or more CCA before transmission.
+	 * The number of CCA attempts to be performed is at most 1 + extra_cca_attempts.
```

Comment:
I would add the information that the CCA attempts are performed back to back and that the transmission starts as soon as idle channel is detected. The following sentence from `ieee802154_ext_ot_config` documentation can be copied here:
> The CCA attempts are performed back-to-back until CCA returns idle channel or until 1 + extra_cca_attempts CCA attempts are performed.